pyomo/core/base/param.py

```diff
@@ -262,6 +262,7 @@ class Param(IndexedComponent, IndexedComponent_NDArrayMixin):
     """
 
     DefaultMutable = False
+    _ComponentDataClass = _ParamData
```

Comment:
The short answer is that Param predated adding `_ComponentDataClass`, so it's use was hard-coded. There is also the issue that the actual entries in the `_data` dict are only `_ParamData` for *mutable* params -- immutable params directly store the native type.  I depated if it was good to define `_ComponentDataClass` (because it isn't always the component data class), but the logic in the expression template classes (especially GetItem) relies on that attribute to determine "what kind" of GetItem expression needs to be created.